fe/fe-core/src/main/java/org/apache/doris/nereids/properties/DistributionSpecHash.java

```diff
@@ -54,6 +54,8 @@ public class DistributionSpecHash extends DistributionSpec {
     private final Set<Long> partitionIds;
     private final long selectedIndexId;
 
+    private final ShuffleFunction shuffleFunction;
```

Comment:
currently we use `shuffleType` to distinguish shuffle functions. `EXECUTION_BUCKETED` is shuffle function used in execution engine and `STORAGE_BUCKETED` is shuffle function used in storage engine. So i think u'd better add a new `ShuffleType ` for spark, rather than add a new attribute in `DistributionSpecHash`

Reply:
I need to add multiple sub types for shuffle type indicating hash functions for it (doris use crc32, spark use murmr32, hive v1, v2, hudi-hash all these are different).
and for each if I add a separate shuffle type then `ChildPropertiesRegulator . visitPhysicalHashJoin()` will become too bloated. 

@morrySnow Can you please suggest a better way to handle the same?

Reply:
Since all these are type of storage-bucketted I'll create a subtype as storage bucket hash type; such that it can be extensible as we add for Hudi, Hive also.
And joins across Spark/Hudi/Hive/Doris can use the left's bucketing hash to deliver correct result.